src/components/common/ColonyHome/ColonyDomainSelector/ColonyDomainSelector.tsx

```diff
@@ -33,18 +33,16 @@ const ColonyDomainSelector = ({
 }: Props) => {
   const { colony, canInteractWithColony } = useColonyContext();
   const { domains } = colony || {};
-
-  // const openEditDialog = useDialog(EditDomainDialog);
-  // const handleEditDomain = useCallback(
-  //   (ethDomainId: number) =>
-  //     openEditDialog({
-  //       ethDomainId,
-  //       colony,
-  //     }),
-  //   [openEditDialog, colony],
-  // );
-
-  const handleEditDomain = () => {};
+  const enabledExtensionData = useEnabledExtensions();
+
+  const openEditDialog = useDialog(EditDomainDialog);
+  const handleEditDomain = (selectedDomainId: number) =>
+    colony &&
```

Comment:
This check is not needed.

The `useColonyContext` hook already does that

Reply:
The alternative pattern we've been using is:

``` js

if(!colony) {
 return null
}
```
which gets typescript off your case about it being maybe undefined.

@jakubcolony Maybe this is actually a legitimate use case for type casting the colony object to be defined if its consumed by the useColonyContext hook, given that we known if it's undefined, we render the 404?

Reply:
If the hook already handles `colony` not being defined, then yeah other than adding checks like this one or the one you suggested to reassure typescript, I think typecasting might be the way in this case.

Other than taking a deeper look into the hook and seeing if we can refactor it so we don't need to typecast that is.